src/span.h

```diff
@@ -18,6 +18,12 @@
 #define ASSERT_IF_DEBUG(x)
 #endif
 
+#if defined(__clang__) && __has_attribute(lifetimebound)
```

Comment:
This seems to fail on gcc 4.8:

```
libtool: compile:  /usr/bin/ccache g++ -m32 -std=c++11 -DHAVE_CONFIG_H -I. -I../src/config -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=2 -I./obj -I./secp256k1/include -DBUILD_BITCOIN_INTERNAL -I/home/travis/build/bitcoin/bitcoin/depends/i686-pc-linux-gnu/share/../include/ -DHAVE_BUILD_INFO -D__STDC_FORMAT_MACROS -D_FILE_OFFSET_BITS=64 -fstack-reuse=none -Wstack-protector -fstack-protector-all -pipe -O2 -fno-extended-identifiers -fvisibility=hidden -c consensus/merkle.cpp  -fPIC -DPIC -o consensus/.libs/libbitcoinconsensus_la-merkle.o

In file included from ./serialize.h:25:0,

                 from ./script/script.h:11,

                 from ./primitives/transaction.h:11,

                 from ./primitives/block.h:9,

                 from ./consensus/merkle.h:10,

                 from consensus/merkle.cpp:5:

./span.h:21:42: error: missing binary operator before token "("

 #if defined(__clang__) && __has_attribute(lifetimebound)

                                          ^

Reply:
Thanks, will break this up into nested if's.